alicloud/data_source_alicloud_mongo_instances.go

```diff
@@ -0,0 +1,263 @@
+						},
+						"replication": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"engine": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"engine_version": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"network_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"lock_mode": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"storage": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"mongos": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"node_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"description": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"class": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"shard": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"node_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"description": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"class": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"storage": {
+										Type:     schema.TypeInt,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAlicloudMongoInstancesRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).ddsconn
+
+	args := dds.CreateDescribeMongoInstancesRequest()
+
+	args.RegionId = getRegionId(d, meta)
+	args.Engine = d.Get("engine").(string)
+	args.DBInstanceType = d.Get("db_type").(string)
+	args.PageSize = requests.NewInteger(PageSizeLarge)
+
+	var dbi []dds.DBInstance
+
+	var nameRegex *regexp.Regexp
+	if v, ok := d.GetOk("name_regex"); ok {
+		if r, err := regexp.Compile(v.(string)); err == nil {
+			nameRegex = r
+		}
+	}
+
+	for {
+		resp, err := conn.DescribeMongoInstances(args)
+		if err != nil {
+			return err
+		}
+
+		if resp == nil || len(resp.DBInstances.DBInstance) < 1 {
+			break
+		}
+
+		for _, item := range resp.DBInstances.DBInstance {
+			if nameRegex != nil {
+				if !nameRegex.MatchString(item.DBInstanceDescription) {
+					continue
+				}
+			}
+			dbi = append(dbi, item)
+		}
+
+		if len(resp.DBInstances.DBInstance) < PageSizeLarge {
+			break
+		}
+
+		args.PageNumber = args.PageNumber + requests.NewInteger(1)
```

Comment:
HI @malferov line 206 is a bug and you can refer to https://github.com/terraform-providers/terraform-provider-alicloud/pull/250/commits/7019687b9931070c00a0281d4f39371990729b26#diff-90d96cd66a3012b469997eb1cbceb593R132

Reply:
fixed